packages/kit/test/apps/basics/svelte.config.js

```diff
@@ -1,6 +1,10 @@
 /** @type {import('@sveltejs/kit').Config} */
 const config = {
 	kit: {
+		output: {
+			preloadStrategy: 'preload-mjs'
```

Comment:
do we really want to switch our largest test app to a non-default strategy? i'd rather have a separate app test this

Reply:
yeah, let's move it into `options`. the tests need updating anyway by the looks of things

Reply:
The recent bug reports around this could have been avoided if your test suite had a test like this:

1. Build the example app.
2. Download the latest `nginx` Docker image and run it, having it serve the build output.
3. Run headless Chrome and open the port served by the Nginx container.
4. Verify that there are no errors in the JavaScript console and that the page has minimal expected behavior (like clicking the button increments the counter).

The Docker orchestration parts could be copied from https://github.com/GeoffreyBooth/js-mjs-mime-type-test/blob/master/test.sh.

Reply:
That's a huge amount of overhead to guard against a very unlikely regression

Reply:
That's simply not going to happen. There are two things we can guard against:

* re-introducing `.mjs`, which would require a maintainer to ignore the 'this is why we don't use `.mjs`' comment that this PR adds
* every conceivable bug that could happen for every conceivable way someone could use SvelteKit, in which case our test suite would take infinite minutes to run, and we would spend all our lives maintaining the test suite inside of the actual code

Hindsight is 20/20 — it would have been nice not to have introduced this breaking change, but the reality of software development is that breakage happens. That's why we have versions!